File: backend/demo1/src/main/java/com/example/demo/entity/RoomReservation.java
package com.example.demo.entity;

import com.example.demo.enums.ReservationStatus;
import jakarta.persistence.*;
import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "room_reservations", uniqueConstraints = {@UniqueConstraint(columnNames = {"room_id","timeslot_id","booking_date","user_id"})})
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class RoomReservation {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    @Enumerated(EnumType.STRING)
    @NotNull(message = "Reservation Status can not be null")
    @Column(nullable = false)
    ReservationStatus reservationStatus;

    @NotNull(message = "Booking date can not be null")
    @Column(nullable = false)
    @FutureOrPresent(message = "You can not choose the past, it needs to be present or future")
    LocalDate bookingDate;

    @CreationTimestamp
    LocalDateTime createdAt;

    @UpdateTimestamp
    LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "timeslot_id")
    private TimeSlot timeSlot;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
}

File: backend/backend-dotnet/Models/RoomReservation.cs
using backend_dotnet.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models;

[Table("room_reservations")]
public class RoomReservation
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public ReservationStatus ReservationStatus { get; set; }

    [Required]
    public DateOnly BookingDate { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public string? AccessCode { get; set; }

    [Column("room_id")]
    public Guid? RoomId { get; set; }

    [ForeignKey(nameof(RoomId))]
    public Room? Room { get; set; }

    [Column("timeslot_id")]
    public Guid? TimeSlotId { get; set; }

    [ForeignKey(nameof(TimeSlotId))]
    public TimeSlot? TimeSlot { get; set; }

    [Column("user_id")]
    public Guid? UserId { get; set; }

    [ForeignKey(nameof(UserId))]
    public User? User { get; set; }
}
